py/mpprint.h

```diff
@@ -39,6 +39,12 @@
 #define PF_FLAG_ADD_PERCENT       (0x100)
 #define PF_FLAG_SHOW_OCTAL_LETTER (0x200)
 
+#if MICROPY_PY_IO
+#    define MP_PRINT mp_sys_stdout_print
```

Comment:
Okay, something like the code below? 

mpprint.h

```
const mp_print_t *mp_default_print(void);
```

mpprint.c

```
const mp_print_t *mp_default_print(void)
{
#if MICROPY_PY_IO
    return &mp_sys_stdout_print;
#else
    return &mp_plat_print;
#endif
}
```

There are also places where there's ``stream_obj`` and ``print`` vs ``mp_plat_print``.
 For example here: https://github.com/micropython/micropython/blob/master/py/modbuiltins.c#L399-L428. How should those be handled?

There is also the alternative to define one of ``mp_plat_print`` and ``mp_sys_stdout_print`` as ``mp_print`` in the header and use that printer everywhere. I mean, should the two printers ever exist at the same time in an application?